File: Code/C#/HVPSAPI/Messages/LiveDataMessage.cs
using Core.Messages.Messages;
using HVPSAPI.DataMemberNames.Messages;
using System.Runtime.Serialization;
using System.Text.Json.Serialization;

namespace HVPSAPI.Messages
{
    [DataContract]
    public class LiveDataMessage : TypedMessageBase
    {
        [JsonPropertyName(LiveDataMessageDataMemberNames.OutputVoltageVolts)]
        [JsonInclude]
        [DataMember(Name = LiveDataMessageDataMemberNames.OutputVoltageVolts)]
        public float OutputVoltageVolts { get; protected set; }
        [JsonPropertyName(LiveDataMessageDataMemberNames.OutputVoltageValueBoundType)]
        [JsonInclude]
        [DataMember(Name = LiveDataMessageDataMemberNames.OutputVoltageValueBoundType)]
        public byte OutputVoltageValueBoundType { get; protected set; }
        [JsonPropertyName(LiveDataMessageDataMemberNames.OutputCurrentAmps)]
        [JsonInclude]
        [DataMember(Name = LiveDataMessageDataMemberNames.OutputCurrentAmps)]
        public float OutputCurrentAmps { get; protected set; }
        [JsonPropertyName(LiveDataMessageDataMemberNames.TotalOutputEnergyJouls)]
        [JsonInclude]   
        [DataMember(Name = LiveDataMessageDataMemberNames.TotalOutputEnergyJouls)]
        public float TotalOutputEnergyJouls { get; protected set; }
        [JsonPropertyName(LiveDataMessageDataMemberNames.FirstStageVoltageValueBoundType)]
        [JsonInclude]
        [DataMember(Name = LiveDataMessageDataMemberNames.FirstStageVoltageValueBoundType)]
        public byte FirstStageVoltageValueBoundType { get; protected set; }
        [JsonPropertyName(LiveDataMessageDataMemberNames.FirstStageVoltageVolts)]
        [JsonInclude]
        [DataMember(Name = LiveDataMessageDataMemberNames.FirstStageVoltageVolts)]
        public float FirstStageVoltageVolts { get; protected set; }
        [JsonPropertyName(LiveDataMessageDataMemberNames.FrequencyHz)]
        [JsonInclude]
        [DataMember(Name = LiveDataMessageDataMemberNames.FrequencyHz)]
        public float FrquencyHz { get; protected set; }
        [JsonPropertyName(LiveDataMessageDataMemberNames.FrequencyHzValueBoundType)]
        [JsonInclude]
        [DataMember(Name = LiveDataMessageDataMemberNames.FrequencyHzValueBoundType)]
        public byte FrequencyHzValueBoundType { get; protected set; }
        [JsonPropertyName(LiveDataMessageDataMemberNames.PeakPrimaryCurrentAmps)]
        [JsonInclude]
        [DataMember(Name = LiveDataMessageDataMemberNames.PeakPrimaryCurrentAmps)]
        public float PeakPrimaryCurrentAmps { get; protected set; }
        [JsonPropertyName(LiveDataMessageDataMemberNames.PeakPrimaryCurrentValueBoundType)]
        [JsonInclude]
        [DataMember(Name = LiveDataMessageDataMemberNames.PeakPrimaryCurrentValueBoundType)]
        public byte PeakPrimaryCurrentValueBoundType { get; protected set; }
        [JsonPropertyName(LiveDataMessageDataMemberNames.PrimaryPowerWatts)]
        [JsonInclude]
        [DataMember(Name = LiveDataMessageDataMemberNames.PrimaryPowerWatts)]
        public float PrimaryPowerWatts { get; protected set; }
        [JsonPropertyName(LiveDataMessageDataMemberNames.TotalPrimaryEnergyJouls)]
        [JsonInclude]
        [DataMember(Name = LiveDataMessageDataMemberNames.TotalPrimaryEnergyJouls)]
        public float TotalPrimaryEnergyJouls { get; protected set; }
        [JsonPropertyName(LiveDataMessageDataMemberNames.MosfetTemperatureDegreesC)]
        [JsonInclude]
        [DataMember(Name = LiveDataMessageDataMemberNames.MosfetTemperatureDegreesC)]
        public float MosfetTemperatureDegreesC { get; protected set; }
        public LiveDataMessage()
            : base()
        {
            Type = MessageTypes.LiveData;
        }
    }
}

File: Code/JavaScript/client/src/messages/LiveDataMessage.js
export default class LiveDataMessage
{
 static toJSON(o){ 
    const r = {};
   r["tpe"]="ld";
    r["a"]=o.outputVoltageVolts;
    r["b"]=o.outputVoltageValueBoundType;
    r["c"]=o.outputCurrentAmps;
    r["d"]=o.totalOutputEnergyJouls;
    r["e"]=o.firstStageVoltageVolts;
    r["f"]=o.firstStageVoltageValueBoundType;
    r["g"]=o.frequencyHz;
    r["h"]=o.frequencyHzValueBoundType;
    r["i"]=o.peakPrimaryCurrentAmps;
    r["l"]=o.peakPrimaryCurrentValueBoundType;
    r["j"]=o.primaryPowerWatts;
    r["k"]=o.totalPrimaryEnergyJouls;
    r["m"]=o.mosfetTemperatureDegreesC;
    r["n"]=o.lowerSnubberDiodeTemperatureDegreesC;
    return r;
 }
 static fromJSON(o){
    const r = {};
    r.outputVoltageVolts=o["a"];
    r.outputVoltageValueBoundType=o["b"];
    r.outputCurrentAmps=o["c"];
    r.totalOutputEnergyJouls=o["d"];
    r.firstStageVoltageVolts=o["e"];
    r.firstStageVoltageValueBoundType=o["f"];
    r.frequencyHz=o["g"];
    r.frequencyHzValueBoundType=o["h"];
    r.peakPrimaryCurrentAmps=o["i"];
    r.peakPrimaryCurrentValueBoundType=o["l"];
    r.primaryPowerWatts=o["j"];
    r.totalPrimaryEnergyJouls=o["k"];
    r.mosfetTemperatureDegreesC=o["m"];
    r.lowerSnubberDiodeTemperatureDegreesC=o["n"];
   return r;
 }
}